src/main/java/rx/internal/operators/DeferredScalarSubscriber.java

```diff
@@ -56,19 +57,31 @@
     /** Value will be emitted. */
     static final int HAS_REQUEST_HAS_VALUE = 3;
     
+    /** prevents multiple terminal emissions. */
+    protected boolean done;
```

Comment:
The problem with this is that not all uses need this flag or the logic. For example, takeLastOne just stores the last value and there is no function involved that could crash. Operators that may crash should individually add these or introduce another abstract class that adds this feature.